esp-hal/src/gpio/mod.rs

```diff
@@ -69,7 +67,11 @@ use crate::{
 #[cfg(touch)]
 pub(crate) use crate::{touch_common, touch_into};
 
+// TODO: Let's keep applications using `gpio::any_pin::{*}` working, as
+// we currently cannot mark specific import path as deprecated.
+#[doc(hidden)]
 pub mod any_pin;
```

Comment:
```suggestion
mod any_pin;
mod dummy_pin;

pub use any_pin::{AnyInputOnlyPin, AnyPin};
pub use dummy_pin::DummyPin;
```

If we make this private, we won't get the weird re-export issue you mentioned. Could you also do the same for `dummy_pin`

Reply:
I managed to reply to issue, instead of PR, where I mentioned that issues I ran into were fixed:
https://github.com/esp-rs/esp-hal/issues/1917#issuecomment-2285487742

Reply:
I did see the comment, but I think its best to make it private and expose the individual pieces separately.